### ISSUE TITLE
Add the warn_continuum_below_threshold checker to all the analysis functions where it's relevant

### ISSUE THREAD

**Issue by eteq** [astropy/specutils#548]
As per the plan in https://github.com/astropy/specutils/issues/535#issuecomment-541783378, 
 after #546 has been done, *if* there's not big performance penalties, the decorator implemented in #538 should be expanded to all the analysis functions that require continuum-subtracted input spectra.

**Comment by eteq** [comment]
Update from #546 - results in https://github.com/astropy/specutils/issues/546#issuecomment-556523323 .

TL;DR is that it's a small (2x or less) effect for functions that operate on the whole spectrum, but potentially very large (~10x) for very large spectra/cubes where you want to do a small operation on only some of them.

So that's an issue.  We may have to pick and choose which the decorators get. So this needs more discussion to decide if we actually want to do it.

I've collated the list of potential places for this decorator.  Below when I say "local" I mean that the measurement is typically in a small spectral region compared to the whole spectrum, whereas "global" means the function operates on the whole spectrum.  Relevant because the "local" ones are the ones where the slow-down could be order-of-magnitude and gets worse with spectrum size (because the continuum-checker is a "global" algorithm), whereas the global ones generally get better with spectrum size (based on #546, of order ~2x slowdown)

- [ ] [centroid](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.centroid.html#specutils.analysis.centroid) - continuum subtraction required, "local" measurement
- [ ] [equivalent_width](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.equivalent_width.html#specutils.analysis.equivalent_width)  - continuum normalization required, "local" measurement
- [ ] [fwhm](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.fwhm.html#specutils.analysis.fwhm)  - continuum subtraction required, "local" measurement
- [ ] [fwzi](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.fwzi.html#specutils.analysis.fwzi)  - continuum subtraction required, "local" measurement
- [ ] [gaussian_fwhm](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.gaussian_fwhm.html#specutils.analysis.gaussian_fwhm)  - continuum subtraction required, "local" measurement
- [ ] [gaussian_sigma_width](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.gaussian_sigma_width.html#specutils.analysis.gaussian_sigma_width)  - continuum subtraction required, "local" measurement
- [ ] [line_flux](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.line_flux.html#specutils.analysis.line_flux)  - continuum subtraction *usually* intended, "local" measurement
- [ ] [find_lines_threshold](https://specutils.readthedocs.io/en/stable/api/specutils.fitting.find_lines_threshold.html#specutils.fitting.find_lines_threshold)  - continuum subtraction required, "global" measurement
- [ ] [fit_lines](https://specutils.readthedocs.io/en/stable/api/specutils.fitting.fit_lines.html#specutils.fitting.fit_lines)  - continuum subtraction sometimes required sometimes not (depends on whether the model has a continuum built-in), "global" *or* "local" measurement

So the question is: which, if any, do we want the continuum checker implemented in as a decorator by default? My first instinct is only `find_lines_threshold`, since that was @camipacifici's original case where there was an issue, and the only definitively global one I found.  But I'm interested in other viewpoints -  @camipacifici (as the inspiration of the original issue), or @keflavich @dhomeier @crawfordsm   (as random potential users), do you have any thoughts here?

**Comment by camipacifici** [comment]
Thank you for the list! The original problem with find_lines_threshold was that it was hard to troubleshoot that the output was wrong because the continuum was not at the expected level (and not for other reasons). Basically, the function was still returning some lines, but not all of them, so the problem seemed to be in the defined threshold and not in the continuum. That is why a check there would be very useful. I do not know how bad the other functions perform in the case of non-appropriate continuum level. Before making a decision, I would check that. If it is obvious that something is wrong, maybe we do not need the decorator. I will check them and report here.

**Comment by camipacifici** [comment]
Here is my report, @eteq . If I do not scale the continuum in the right way:
- `centroid` gives the wrong answer, but it is not obviously wrong. A local check would be useful.
- `equivalent_width` gives an obviously wrong answer. No need for a check.
- `fwhm`  gives the wrong answer, but it is not obviously wrong. A local check would be useful.
- `fwzi` same as `fwhm`.
- `line_flux` gives an obviously wrong answer. No need for a check.
- `find_lines_threshold` gives an answer, but it is hard to tell if the list of lines is complete and it actually looks incorrect. It is not obvious if the problem is in the continuum or in the input parameters. A check would be useful.
- `fit_lines` gives an obviously wrong answer, at least using a Gaussian model. The problem could still be related to the input parameters of the model. A check would be useful.

`gaussian_fwhm` and `gaussian_sigma_width` are returning Nan when using the correctly subtracted spectrum. I need to investigate these two a little more.

**Comment by eteq** [comment]
@camipacifici - hmm. Ok so I think the easy one is `find_lines_threshold` - we should leave that in since it doesn't have a big performance effect and it's definitely useful.  Currently I guess `find_lines_derivative` *also* has it, but maybe we don't want that since `derivative` is based.

`fit_lines` is a bit iffier, but my thinking is that maybe we don't need the decorator there on the theory that the user can (if they want to) include the continuum in the model itself.  That means sometimes the decorator would give the *wrong* answer if we have it on there (e.g., a gaussian + continuum model should *not* have the continuum subtracted), so a bit more thought suggests we shouldn't have it there.

But `centroid` and `fwhm`/`fwzi`/`gaussian_fwhm` are the hard ones for the decorator then, since from the above they would be useful but have serious performance impacts...
